backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/ObservationHandler.java

```diff
@@ -230,11 +231,11 @@ public ObservationHandler(Executor executor, FutureJdbi jdbi) {
         handle -> {
           // Delete from keyvalue
           var sql =
-              "delete from keyvalue where id in "
-                  + "(select keyvaluemapping_id from observation where entity_name=:entity_name and field_type=:field_type and experiment_run_id=:run_id)";
+              "delete o, k from keyvalue AS k inner join observation AS o ON o.keyvaluemapping_id=k.id "
+                  + "where o.entity_name=:entity_name and o.field_type=:field_type and o.experiment_run_id=:run_id ";
```

Comment:
In the old query we can not delete the keyvalue because there is a foreign key constraint with observation. so removed implementation of query doesn't work. So added new query after testing and new query is working fine now.

Reply:
delete+join is likely not going to work with all SQL versions. can we just remove the foreign key instead? foreign keys are evil and I don't want to make our code more complicated and incompatible to work around them

Reply:
Simplify query by following flow
- Use transaction
- Get observation and keyvalue  ids in map
- Remove foreign key constraint by setting null in mapping column
- Delete keyvalue
- Delete observation